fipy/tools/logging/environment.py

```diff
@@ -55,6 +55,37 @@ def pip_info(python="python"):
 
     return json.loads(stdout)
 
+def nix_info():
+    """Collect information about nix environment.
+
+    Returns
+    -------
+    dict
+        Result of `nix derivation show .#fipy `.
+    """
+    info = {}
+
+    # Better output would be from
+    #
+    #     $ nix-store -q --tree $(nix-store --realize $(nix eval --raw .#fipy.drvPath))
+    #
+    # However, this is difficult to execute and can require another build.
+    # Also doesn't return json.
+
+    p = subprocess.Popen(
+        [
+         "nix",
+         "derivation",
+         "show",
+         ".#fipy"
+        ], stdout=subprocess.PIPE)
+
+    stdout, _ = p.communicate()
+    stdout = stdout.decode('ascii')
+
+    info["nix_info"] = json.loads(stdout)
```

Comment:
Arguably, this should just return `json.loads(stdout)`, rather than nesting it in a dict. `conda_info()` is a little different because it collects the results of both `conda info` and `conda env export`. One option is that each of these routines should return a dict containing one or more info["???_info"] and then have `fipy.__init__.py` call `_fipy_environment.update(environment.???_info())` on each of them.

Reply:
I'll let you decide on that.